File: src/SkyBlock/enchants/sword/Blind.php
<?php


namespace SkyBlock\enchants\sword;


use pocketmine\entity\effect\EffectInstance;
use pocketmine\entity\effect\VanillaEffects;
use pocketmine\event\entity\EntityDamageEvent;
use pocketmine\player\Player;

class Blind extends BaseMeleeEnchant {

    static int $id = 101;

    public function onActivation(Player $player, Player $attacker, EntityDamageEvent $ev, int $enchantmentlevel) : void {
        $this->sendActivation($attacker, "§cStruck by §bBlind §cEnchant!");
        $this->sendActivation($player, "§bBlind §aActivated!");
        $effect = new EffectInstance(VanillaEffects::BLINDNESS(), $this->getDuration($enchantmentlevel), 5, false);
        $attacker->getEffects()->add($effect);
    }
}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/sword/Blind.java
package dev.skyblock.enchants.sword;

import dev.skyblock.enchants.BaseMeleeEnchant;
import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;
import java.util.concurrent.ThreadLocalRandom;

public class Blind extends BaseMeleeEnchant {
    public static final int ID = 101;

    public Blind() {
        super(ID, "Blind", 5);
    }

    @Override
    public boolean isApplicableTo(Player holder, int level) {
        return ThreadLocalRandom.current().nextInt(1, 26) == 1;
    }

    @Override
    public void onActivation(Player victim, Player attacker, EntityDamageByEntityEvent ev, int level) {

    }
}
